packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -106,18 +119,50 @@ class AttachCommand extends FlutterCommand {
       : null;
 
     Uri observatoryUri;
+    bool ipv6 = false;
+    bool attachLogger = false;
     if (devicePort == null) {
-      ProtocolDiscovery observatoryDiscovery;
-      try {
-        observatoryDiscovery = ProtocolDiscovery.observatory(
-          device.getLogReader(),
-          portForwarder: device.portForwarder,
+      if (device is FuchsiaDevice) {
+        attachLogger = true;
+        final String module = argResults['module'];
+        if (module == null) {
+          throwToolExit('\'--module\' is requried for attaching to a Fuchsia device');
+        }
+        ipv6 = _isIpv6(device.id.split('%').first);
```

Comment:
Consider moving the .split('%') down inside of _isIpv6, and note that it is needed due to https://github.com/dart-lang/sdk/issues/29456